bigframes/core/blocks.py

```diff
@@ -2488,6 +2488,11 @@ def join(
             )
             if result is not None:
                 return result
+
+            # For block identify joins with null indices, perform cross join
```

Comment:
This doesn't seem desirable. If df1 is n rows and df2 is m rows, won't this end up with n x m rows?